crates/wasm_module/src/sections.rs

```diff
@@ -194,7 +194,7 @@ pub fn update_section_size<T: SerialBuffer>(buffer: &mut T, header_indices: Sect
  *
  *******************************************************************/
 
-#[derive(PartialEq, Eq, Debug)]
+#[derive(PartialEq, Eq, Debug, Clone)]
```

Comment:
maybe the `param_types` should be a slice? not sure if that works with all the other code, but at least in theory, once created the (number of) arguments should not change right?

Reply:
Yeah that should be possible, maybe I'll update that some time. I don't think there's any code that _tries_ to change the number of args anyway though. Is there some other benefit?
Oh.. cloning it is probably cheaper I suppose.
Yeah the only thing cloning it is a test. I guess I was being a little sloppy by doing it this way, I could have copy and pasted code in the test instead.
I can do that in the next PR and get rid of the Clone.

Reply:
Yeah good catch on this. Adding a Clone trait just for test code was a bit messy.
There's a fair bit of code that would need to be updated to add it for real so it seems like it makes more sense to leave the Vec and change the test.